pkgs/native_assets_cli/lib/src/config.dart

```diff
@@ -404,6 +406,168 @@ class BuildOutputBuilder extends HookOutputBuilder {
       syntax.BuildOutput.fromJson(super._syntax.json);
 }
 
+extension DataAssetsDirectoryExtension on BuildOutputBuilder {
+  /// Extension on [BuildOutput] to handle data asset directories and files.
+  ///
+  /// This extension provides a convenient way for build hooks to add
+  /// [DataAsset] dependencies from one or more directories or individual files.
+  ///
+  /// If any specified path does not exist, a [FileSystemException] is thrown.
+  /// Any error during the directory listing is caught and rethrown with
+  /// additional context.
+  Future<void> addDataAssetDirectories(
+    List<String> paths, {
+    required BuildInput input,
+  }) async {
+    final packageName = input.packageName;
+    final packageRoot = input.packageRoot;
+    for (final path in paths) {
+      final resolvedUri = packageRoot.resolve('$packageName/$path');
```

Comment:
Why do we have `$packageName/` here? Wouldn't we simply want to resolve `$path`?

Reply:
Files created in the tests were showing up under $packageName, is there something else moving the assets to this path?

Reply:
How can test_data/simple_data_asset be tested? It depends on a nonexistent dart:asset and ByteAsset. 

Pushed a fix for name that mirrors that example, we still must resolve the relative $packageName/$path to find the file though. Do you want this to be refactored for compatibility with system absolute paths or relative only?